src/front/listener.rs

```diff
@@ -25,6 +26,29 @@ pub struct ListenerConfig {
     pub timeout: u64,
 }
 
+/// Specifies metadata associated with a connection, if any.
+#[derive(Copy, Clone, Debug)]
+pub enum ConnectionMetadata {
+    NoMetadata,
+}
+
+/// Represents a connection to a single client. Contains a stream, used for communication with the
+/// client, and some metadata associated with the connection that might be useful elsewhere (i.e.
+/// authentication, etc).
+pub struct Connection {
+    pub stream: Box<dyn ReadWrite + Send>,
+    pub metadata: ConnectionMetadata,
+}
```

Comment:
I think it's more idiomatic instead of having an enum with a variant `EnumType::NoSomething`, to actually use `Option<EnumType>`. You can probably just create an "empty" `UnixSocket` variant that we flesh out when we actually need that metadata.

Reply:
I completely agree. Will change this. :)